### ISSUE TITLE
Make cosmo-ghg-11km-test more general

### ISSUE THREAD

**Issue by efmkoene** [C2SM/processing-chain#19]
Currently, the `cosmo-ghg-11km-test` case works well for the integrated Jenkins suite with GPU, but it requires more work for other users. I think a smoother experience is possible.

The issue shows up in two ways:
1. We are limited to using the Jenkins testsuite build of int2lm and cosmo, for example it is hardcoded that `cosmo_bin = os.path.join(chain_src_dir, 'src', 'cosmo-ghg', 'cosmo', 'ACC','cosmo_gpu')` and . Is there no way to point to a C2SM maintained version of these codes if the user is not Jenkins?
2. Some things are hardcoded w.r.t. GPUs that fails when we switch to CPU, for example:
   *    The number of nodes goes to zero (and thus fails to run COSMO) due to `cosmo_np_tot = int(cosmo_np_x * cosmo_np_y / ntasks_per_node) + cosmo_np_io` = int( 4 * 3 / 36) + 0=0. 
   *    We need to change `cosmo_bin` to use some variation of the CPU build rather than the GPU version
   *    In cosmo_INPUT_DYN.cfg we need lcpp_dycore=.FALSE. instead of lcpp_dycore=.TRUE.

If issue 1 cannot be fixed, I'd still like to address issue 2, i.e., to put some simple statements behind `if constraint=='GPU' ... elif constraint=='CPU' ...`, for a more generally applicable testcase. Is that OK for you @mjaehn ? And if so, should I do this in the refactoring PR or at a later point?

**Comment by mjaehn** [comment]
Hi @efmkoene,

Thanks for your suggestions to improve the processing chain - this is always welcome!

1. The test cases are just meant to do automatic tests, that's why there are hardcoded paths to the executables that align with the previously executed jenkins scripts. Now, I don't think if I got your question right, because if you want to manually launch this test case, you would just use `./jenkins/scripts/jenkins.sh` - or manually execute the scripts in `./jenkins/scripts/*.sh`. In either case, there is no need to change cosmo_bin in the `config.py` file. The design of the new `spack-c2sm` explicitly states that each user has their own spack instance and thus installs their own software versions. Only certain dependencies are provided by C2SM in an upstream instance (for example `eccodes`) to speed up the installation process. If a user has their own processing chain case, they will anyway specify their own location of their executable.
2. If you want to test the chain with the cpu version of cosmo, I suggest to create a new case based on the gpu one and adapt the values for the number of nodes etc. Calculations will be removed anyway from the `config.py` within the [refactoring](https://github.com/C2SM/processing-chain/pull/18) subtask as it will be transformed to a `config.yaml` file. The calculation of `cosmo_np_tot` should then be moved to the  `cosmo` job itself.

I hope this clarifies the intention behind those test cases. If not, let me know and we will figure out 😃 

 @efmkoene Would you be up to creating a cosmo-ghg cpu test case?

**Comment by efmkoene** [comment]
Right, I didn't know the testcase was just for the integrated testing environment. I thought it existed also for human users to show what the code can do; this is the code I ran on the first day when I started working here. 

The reason I raised this question in the past is because I wanted to test the processing-chain for a pull request review at some point; but I didn't have GPU nodes, and it took some effort to make it work with CPU nodes.

I agree with your suggestion, that perhaps the best thing to do then is to make an additional test-case, for humans, that can handle either cpu or gpu versions with some simple if-statements. Sounds like I should wait until the refactoring subtask is done, though.

**Comment by efmkoene** [comment]
(And one more thing, I don't want to open an issue for this, but the README instructions explain how to compile COSMO, but to run any of the COSMO cases, you also need to compile INT2LM. So the request is: can you add the INT2LM compilation instructions into the README too?)

**Comment by mjaehn** [comment]
Thanks for pointing out the README. I am aware that due to the recent changes it is somehow outdated. It is on my TODO list in the [refactoring PR](https://github.com/C2SM/processing-chain/pull/18).
